tests/__mocks__/psElevateTokenHandler.js

```diff
@@ -0,0 +1,254 @@
+     * @return Promise A token promise
+     */
+    requestToken({
+                     iframe,
+                     handleError,
+                     resolveToken,
+                     eventAction,
+                     tokenizeParameters,
+                 } = {}) {
+        if (isNull(iframe)) {
+            return;
+        }
+
+        const tokenPromise = new Promise((resolve, reject) => {
+
+            const timer = setTimeout(() =>
+                    reject(handleError({
+                        error: this.labels.tokenRequestTimedOut,
+                        isObject: false,
+                    })),
+                TOKENIZE_TIMEOUT_MS,
+            );
+
+            this.tokenCallback = message => {
+                clearTimeout(timer);
+
+                if (message.error) {
+                    reject(handleError(message));
+
+                } else if (message.token) {
+                    if (resolveToken) {
+                        resolve(resolveToken(message.token));
+                    } else {
+                        resolve(message.token);
+                    }
+                }
+            };
+
+        });
+
+        const message = {
+            action: eventAction,
+            params: tokenizeParameters,
+        };
+        this.sendIframeMessage(iframe, message, this._visualforceOrigin);
+
+        return tokenPromise;
+    }
+
+    setPaymentMethod(iframe, paymentMethod, handleError, resolveSetPaymentMethod) {
+        if (isNull(iframe)) {
+            return;
+        }
+
+        const setPaymentMethodPromise = new Promise((resolve, reject) => {
+            this.tokenCallback = message => {
+                if (message.error) {
+                    reject(handleError(message));
+                } else {
+                    resolve(resolveSetPaymentMethod());
+                }
+            };
+        });
+
+        const message = {
+            action: SET_PAYMENT_METHOD_EVENT_ACTION,
+            paymentMethod: paymentMethod
+        };
+        this.sendIframeMessage(iframe, message, this._visualforceOrigin);
+
+        return setPaymentMethodPromise;
+    }
+
+    mount(iframe, paymentMethod, handleError, resolveMount) {
+        if (isNull(iframe)) {
+            return;
+        }
+
+        const mountPromise = new Promise((resolve, reject) => {
+            this.tokenCallback = message => {
+                if (message.error) {
+                    reject(handleError(message));
+                } else {
+                    resolve(resolveMount());
+                }
+            };
+        });
+
+        const message = {
+            action: MOUNT_IFRAME_EVENT_ACTION,
+            paymentMethod: paymentMethod
+        };
+        this.sendIframeMessage(iframe, message, this._visualforceOrigin);
+
+        return mountPromise;
+    }
+
+
+    sendIframeMessage(iframe, message, targetOrigin) {
+        const reply = mockGetIframeReply(iframe, message, targetOrigin);
```

Comment:
Here's where we insert the mock for getting a message back from the iframe. Happens synchronously. Would like to be able to provide a mock for the iframe itself instead eventually.